Local variable `page`in unit test conflicts with API type `page`

[Dialogflow CX v3beta1](https://github.com/googleapis/googleapis/tree/master/google/cloud/dialogflow/cx/v3beta1)

Dialogflow has a `page` type that causes two of the generated unit tests to fail because of a name conflict.

https://github.com/googleapis/gapic-generator-python/blob/b2e52746c7ce4b983482fb776224b30767978c79/gapic/templates/tests/unit/gapic/%25name_%25version/%25sub/test_%25service.py.j2#L803

```py
def test_list_pages_pages():
    client = PagesClient(credentials=credentials.AnonymousCredentials,)

    # Mock the actual call within the gRPC stub, and fake the request.
    with mock.patch.object(type(client._transport.list_pages), "__call__") as call:
        # Set the response to a series of pages.
        call.side_effect = (
            page.ListPagesResponse(  # page as a type
                pages=[page.Page(), page.Page(), page.Page(),], next_page_token="abc",
            ),
            page.ListPagesResponse(pages=[], next_page_token="def",),
            page.ListPagesResponse(pages=[page.Page(),], next_page_token="ghi",),
            page.ListPagesResponse(pages=[page.Page(), page.Page(),],),
            RuntimeError,
        )
        pages = list(client.list_pages(request={}).pages)
        for page, token in zip(pages, ["abc", "def", "ghi", ""]):  # page used as local var
            assert page.raw_page.next_page_token == token
```

```py
@pytest.mark.asyncio
async def test_list_pages_async_pages():
    client = PagesAsyncClient(credentials=credentials.AnonymousCredentials,)

    # Mock the actual call within the gRPC stub, and fake the request.
    with mock.patch.object(
        type(client._client._transport.list_pages),
        "__call__",
        new_callable=mock.AsyncMock,
    ) as call:
        # Set the response to a series of pages.
        call.side_effect = (
            page.ListPagesResponse(
                pages=[page.Page(), page.Page(), page.Page(),], next_page_token="abc",
            ),
            page.ListPagesResponse(pages=[], next_page_token="def",),
            page.ListPagesResponse(pages=[page.Page(),], next_page_token="ghi",),
            page.ListPagesResponse(pages=[page.Page(), page.Page(),],),
            RuntimeError,
        )
        pages = []
        async for page in (await client.list_pages(request={})).pages:
            pages.append(page)
        for page, token in zip(pages, ["abc", "def", "ghi", ""]):
            assert page.raw_page.next_page_token == token
```

```
E           UnboundLocalError: local variable 'page' referenced before assignment

tests/unit/gapic/dialogflowcx_v3beta1/test_pages.py:622: UnboundLocalError
```

--- a/gapic/ads-templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2
+++ b/gapic/ads-templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2
@@ -522,8 +522,8 @@ def test_{{ method.name|snake_case }}_pages():
             RuntimeError,
         )
         pages = list(client.{{ method.name|snake_case }}(request={}).pages)
-        for page, token in zip(pages, ['abc','def','ghi', '']):
-            assert page.raw_page.next_page_token == token
+        for page_, token in zip(pages, ['abc','def','ghi', '']):
+            assert page_.raw_page.next_page_token == token
 {% elif method.lro and "next_page_token" in method.lro.response_type.fields.keys() %}
 def test_{{ method.name|snake_case }}_raw_page_lro():
     response = {{ method.lro.response_type.ident }}()

--- a/gapic/templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2
+++ b/gapic/templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2
@@ -838,8 +838,8 @@ def test_{{ method.name|snake_case }}_pages():
             RuntimeError,
         )
         pages = list(client.{{ method.name|snake_case }}(request={}).pages)
-        for page, token in zip(pages, ['abc','def','ghi', '']):
-            assert page.raw_page.next_page_token == token
+        for page_, token in zip(pages, ['abc','def','ghi', '']):
+            assert page_.raw_page.next_page_token == token
 
 @pytest.mark.asyncio
 async def test_{{ method.name|snake_case }}_async_pager():
@@ -928,10 +928,10 @@ async def test_{{ method.name|snake_case }}_async_pages():
             RuntimeError,
         )
         pages = []
-        async for page in (await client.{{ method.name|snake_case }}(request={})).pages:
-            pages.append(page)
-        for page, token in zip(pages, ['abc','def','ghi', '']):
-            assert page.raw_page.next_page_token == token
+        async for page_ in (await client.{{ method.name|snake_case }}(request={})).pages:
+            pages.append(page_)
+        for page_, token in zip(pages, ['abc','def','ghi', '']):
+            assert page_.raw_page.next_page_token == token
 {% elif method.lro and "next_page_token" in method.lro.response_type.fields.keys() %}
 def test_{{ method.name|snake_case }}_raw_page_lro():
     response = {{ method.lro.response_type.ident }}()

